tests/unit/test_configuration.py

```diff
@@ -1,7 +1,67 @@
+# TODO: type ignores and crawlee_storage_dir
+# https://github.com/apify/crawlee-python/issues/146
+
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
 from crawlee.configuration import Configuration
+from crawlee.http_crawler import HttpCrawler, HttpCrawlingContext
+from crawlee.memory_storage_client._memory_storage_client import MemoryStorageClient
+from crawlee.service_container import get_configuration, set_storage_client
+
+if TYPE_CHECKING:
+    from pathlib import Path
 
 
 def test_global_configuration_works() -> None:
-    assert Configuration.get_global_configuration() is Configuration.get_global_configuration()
+    assert (
+        Configuration.get_global_configuration()
+        is Configuration.get_global_configuration()
+        is get_configuration()
+        is get_configuration()
+    )
+
+
+async def test_storage_not_persisted_when_disabled(tmp_path: Path) -> None:
+    config = Configuration(
+        persist_storage=False,
+        write_metadata=False,
+        crawlee_storage_dir=str(tmp_path),  # type: ignore
+    )
+    storage_client = MemoryStorageClient(config)
+    set_storage_client(storage_client, force=True)
```

Comment:
Sorry, could you elaborate it a bit? I am not sure I know what you have in mind.

If you are referring to passing options to the `MemoryStorageClient`, you can now provide them directly:

```python
    storage_client = MemoryStorageClient(
        persist_storage=False,
        write_metadata=False,
        storage_dir=str(tmp_path),
    )
    set_storage_client(storage_client, force=True)
```

Reply:
Yes, but in JS you can do this:

```ts
new BasicCrawler({
  configuration: new Configuration({
    storageClient: SQLiteStorageClientIDontKnow()
  })
})
```

...and it just works. I'm not advocating this exact API, it's really not that great. My point is, the `*Crawler` classes are basically the entrypoint to Crawlee. If we could have a way to tweak the "dependencies" using their constructors alone, it could make this functionality more discoverable... Which would be nice, given that users evidently struggle with the default filesystem-backed memory storage often :slightly_smiling_face:

Reply:
Sure, `BasicCrawler` will accept storage client as an input arg.